ibis/expr/rules.py

```diff
@@ -424,11 +424,17 @@ def table(arg, *, schema=None, **kwargs):
 
 @validator
 def column_from(name, column, *, this):
-    if not isinstance(column, (str, int)):
-        raise com.IbisTypeError(
-            f"value must be an int or str, got {type(column).__name__}"
-        )
-    return getattr(this, name)[column]
+    if isinstance(column, ir.AnyColumn) and hasattr(column.op(), "name"):
+        maybe_column = column.op().name
+        if column.equals(getattr(this, name)[maybe_column]):
+            return column
```

Comment:
Probably should give a more specific error message in the `else:` clause like `passed column is not a column in {getattr(this, name)}`

Reply:
This should still be a `com.IbisTypeError`?